src/loaders/common.c

```diff
@@ -258,6 +258,55 @@ int libxmp_test_name(uint8 *s, int n)
 	return 0;
 }
 
+int libxmp_copy_name_for_fopen(char *dest, const char *name, int n)
+{
+	int converted_colon = 0;
+	int i;
+
+	/* libxmp_copy_adjust, but make sure the filename won't do anything
+	 * malicious if it's given to fopen. This should only be used on song files.
+	 */
+	if (!strcmp(name, ".") || strstr(name, "..") || name[0] == '\\' || name[0] == '/')
```

Comment:
this should also check if the first character is a colon, because otherwise (due to colon version) you can still sneak in an absolute path.

Reply:
I updated this to filter out anything with `:` as the first character (can't imagine that was ever valid even on Amiga) and also to limit the colon converstion to i>=1.

Reply:
This looks good.  OK to merge now?